app/helpers/holdings_helper.rb

```diff
@@ -111,7 +118,7 @@ def electronic_access_url(enable_proxy, proxy, url)
   end
 
   def library(item)
-    @libraries[LIBRARIES[item[:location]]]
+    Library.find_by!(short_code: LIBRARIES[item[:location]])
```

Comment:
What do we want to happen if we can't map the current item[:location] to an institution and then further if we can't find a Library?

Reply:
See above - the way cataloguing works, there should be no items without a location, and no location without a larger institution. If that happens, it should fail gracefully and a NEOS library can report a cataloguing correction.

Reply:
@redlibrarian Agreed and this reminds me that Cataloguing should develop something in BCA that can deliver reports of wacko items that have null libraries/locations (impossible for a human to create, as you stated, but I think can happen as a very rare, unintended/undetected consequence of automated tasks)